Create app skeleton #01

Forestpw opened the pull request:
Following changes have been implemented as per the acceptance criteria:

- Install fonts via import in style.css
- reset default browser style with import-normalize
- add global styles
- add router with routes / and /search
- add folder structure

Forestpw committed:
style: Add Bitter and Montserrat fonts to stylesheet

Create a folder for the stylesheet called Style. Within the folder a style.css
was created and modified with an import for the Google Fonts Bitter and Montserrat.

Resolves: #01

Forestpw committed:
style: Add reset default browser styles

Add normalize.css to normalize all default css across all browser css presets.

Resolves: #01

Forestpw committed:
feat: Add folder Components and Search.js within

The start of the components folder that will hold all components
and routes for the application. Within the first route Search.js
was created with a basic template.

Resolves: #01

Forestpw committed:
feat: Add react-router-dom to package.json

react-router-dom will allow the application to route
between specific paths.

Resolves: #01

Forestpw committed:
feat: Add routes within the App.js

Create the basic template for routes within the App.js
file. This will be expanded upon later but for now a template
will suffice.

Resolves: #01

Forestpw committed:
style: Add global styles

First add of the global styles for the stylesheet
style.css

Resolves: #01

Forestpw committed:
style: Refactor normalize.css to single location

normalize.css did not need its own css file and instead could
be imported in the style.css file within a single line.

Resolves: #01

mentor-tara commented:
Great job :rocket: I'm a bit busy at the moment but I'll leave a review soon.

In the meantime **I'd suggest you start with the next task already. Even if you just check out a new branch** where you can continue to work on later. This is a great opportunity to learn professional Git workflows. 

[Head back to the course material to learn how to proceed.](https://profy.dev/project/reddit-timer/while-waiting)

PS: Of course we all know I'm not busy. I'm a bot :robot: But let's pretend that I am to have a more realistic team environment. We're here to practice after all, right? :smile:[](ckkp04mmw2vuk0b42idpakyah)

mentor-tara reviewed (changes requested):
Before I approve this PR I'd like to share a few tips with you.

**Please go through the list below and review your own code first.** Imagine you're the reviewer or a person who didn't see the code before. Ask yourself if everything is clearly written and understandable. Are components and variables properly named?

Reviewing your own code with this mindset is an important practice that will help you learn a lot and write high-quality code.

To make it a bit easier for you here are the tips I promised. Some have a link :link: attached where you can find more information. I'd recommend reading the **bold tips** in any case. Feel free to check the boxes when you think your code fits.

**General:**
- [x] **Use yarn instead of npm** [:link:](https://profy.dev/project/reddit-timer/review-tips/use-yarn)
- [x] **Use a common project structure** [:link:](https://profy.dev/project/reddit-timer/review-tips/file-structure)

Once you're done with reviewing your own code feel free to check the [example implementation](https://profy.dev/project/reddit-timer/code-examples/create-app-skeleton-d137af0d-ce79-4289-ae9b-df6f0444a380). **But remember: It's an important exercise to review your code first on your own.**

When you think this PR is ready to be merged just let me know by requesting another review. I'll approve right away.[](ckkp0fo0g2w0w0b422db33fjd)

Forestpw committed:
style: Add fonts the right way

This is the proper way to download the fonts since
we will be using styled-components rather than a css.

Resolves: #01

Forestpw committed:
feat: Add the propery dependecies for setup

Add dependecies for react-router-dom, styled-components,
and styled-normalize.

Resolves: #01

Forestpw committed:
style: Remove of style.css

Remove of style.css since we will be using
styled-components instead.

Resolves: #01

Forestpw committed:
refactor: Remove of base react images

Removal of the standard react images with an update to
the pages description to better match what the site is
actually about.

Resolves: #01

Forestpw committed:
refactor: Update Routes found in App.js

Update App.js to have better readability for the routes
by making them on a single line rather than span across
three lines. Also Normalize is refactored to be imported
from the dependecy rather than from a css file.

Resolves: #01

Forestpw committed:
style: Add global style to app

This is the proper way we should have added global
styles to the app with styled-components rather than
use a css file.

Resolves: #01

Forestpw committed:
style: Add theme to App.js

Create theme within the theme.js and imported
it within the App.js and wrapped everything within
ThemeProvider.

Resolves: #01

mentor-tara reviewed (approved):
Great work! :tada::tada::tada: You can merge the PR now.[](ckkqnj5io3o540b425ilxp80y)

jkettmann commented:
Looks really good. I like that you made use of the example implementation. Keep on the good work 😄
